resources/main.mjs

```diff
@@ -17,17 +17,49 @@ class MainBenchmarkClient {
     _hasResults = false;
     _developerModeContainer = null;
     _metrics = Object.create(null);
+    _steppingPromise = null;
+    _steppingResolver = null;
 
     constructor() {
         window.addEventListener("DOMContentLoaded", () => this.prepareUI());
         this._showSection(window.location.hash);
     }
 
     start() {
-        if (this._startBenchmark())
+        if (this._isStepping())
+            this._clearStepping();
+        else if (this._startBenchmark())
             this._showSection("#running");
     }
 
+    step() {
+        const currentSteppingResolver = this._steppingResolver;
+        this._steppingPromise = new Promise((resolve) => {
+            this._steppingResolver = resolve;
+        });
+        if (this._isStepping())
+            currentSteppingResolver();
+        if (!this._isRunning) {
+            this._startBenchmark();
+            this._showSection("#running");
+        }
+    }
+
+    _clearStepping() {
+        this._steppingResolver();
+        this._steppingPromise = null;
+        this._steppingResolver = null;
```

Comment:
Shouldn't we be clearing these two states before calling `_steppingResolver`?

Reply:
sounds good.